src/test/test-time-util.c

```diff
@@ -870,6 +864,66 @@ TEST(parse_timestamp) {
                 test_parse_timestamp_one("30minutes ago", USEC_PER_MINUTE, now_usec - 30 * USEC_PER_MINUTE);
 }
 
+TEST(parse_timestamp) {
+        test_parse_timestamp_impl(NULL);
+}
+
+static void test_parse_timestamp_with_tz_one(const char *name1, const char *name2) {
+        _cleanup_free_ char *buf = NULL, *tz = NULL;
+        const char *name, *saved_tz;
+
+        if (name2)
+                assert_se(buf = path_join(name1, name2));
+        name = buf ?: name1;
+
+        if (!timezone_is_valid(name, LOG_DEBUG))
+                return;
+
+        log_info("/* %s(%s) */", __func__, name);
+
+        saved_tz = getenv("TZ");
+
+        assert_se(tz = strjoin(":", name));
+        assert_se(setenv("TZ", tz, 1) >= 0);
+        tzset();
+        log_debug("%s: tzname[0]=%s, tzname[1]=%s", tz, strempty(tzname[0]), strempty(tzname[1]));
+
+        test_parse_timestamp_impl(tz+1);
+
+        assert_se(set_unset_env("TZ", saved_tz, true) == 0);
+        tzset();
+}
+
+TEST(parse_timestamp_with_tz) {
+        if (!slow_tests_enabled())
+                return (void) log_tests_skipped("slow tests are disabled");
+
+        _cleanup_closedir_ DIR *dir = opendir("/usr/share/zoneinfo");
```

Comment:
maybe use get_timezones() instead?

if you do things shouldn't be "slow" anymore, either

Reply:
ah, the slowness comes from the forking? ignore the comment about the "slow" then. 

but get_timezones() might still be preferable to use?